app/models/foreman_tasks/concerns/action_triggering.rb

```diff
@@ -35,13 +35,17 @@ def destroy
         dynflow_task_wrap(:destroy) { super }
       end
 
-      def update_attributes(*args)
+      def update(*args)
+        self.attributes = args[0]
```

Comment:
Why? It's not intended here to do anything else than the `dynflow_task_wrap`. Unless it's really related to the functionality of the `ActionTriggering`, it should be somewhere else.

Reply:
This does fix [#23837](https://projects.theforeman.org/issues/23837), and here's why: we are using `host.<attribute>_changed?` [here](https://github.com/Katello/katello/blob/master/app/models/katello/host/subscription_facet.rb#L253) to check whether we need to trigger the `update_action`. Those methods do not work if you use `host.update(attributes)` because the attributes aren't updated in memory on the host, so the method always returns false. Instead, we must use `host.attributes = attributes; host.save`.

This controller works without this line: [app/controllers/api/v2/hosts_controller.rb#L161](https://github.com/theforeman/foreman/blob/develop/app/controllers/api/v2/hosts_controller.rb#L161)
This one doesn't: [app/controllers/hosts_controller.rb#L122](https://github.com/theforeman/foreman/blob/develop/app/controllers/hosts_controller.rb#L122)

This is **one** solution, but it seems pretty hacky to me. I'd welcome other suggestions. :)

Reply:
What about doing this in https://github.com/theforeman/foreman/blob/develop/app/controllers/hosts_controller.rb#L122? It seems like a Rails bug, but we should update the usage of the `@host.update` in Foreman, if the `host.<attribute>_changed?` doesn't work with the `update` method.

```ruby
@host.attributes = attributes
if @host.save
  #...
```

Reply:
I'm fine with changing it in Foreman. What's one more PR for this issue? ;)

Reply:
Though, if it's not a bug in Rails, we may want to include some more documentation with this ActionTriggering module to inform people that it must be used with the two-step save. Putting the attributes on the host here in this repo removes that usage caveat.

Reply:
Oh, I understand the issue now: I agree this is actually an issue here then.

I would prefer avoiding calling args[0] in favor of usign *args with `assign_attributes`
```
assign_attributes(*args)
```

Also, please explanatory comment why we are doing this in the first place, for future us still being able to understand why.